commit 41cf089f6079a3a2d564a2f76456a4d436deb780 (refs/changes/71/2271/3)
Author: Jared Morrow <jm@joyent.com>
Date:   2017-09-11T13:17:59-06:00 (2 years, 1 month ago)
    
    MANTA-3357 manta-scripts update script repos.sh no longer works with modern git
    Reviewed by: Kody A Kantor <kody.kantor@gmail.com>
    Approved by: Kody A Kantor <kody.kantor@gmail.com>

diff --git a/update/repos.sh b/update/repos.sh
index fd24a20..6537efc 100755
--- a/update/repos.sh
+++ b/update/repos.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 ###############################################################################
@@ -19,6 +19,8 @@
 # with repos, you can fix and safely rerun.
 ###############################################################################
 
+set -o xtrace
+
 REPOS=(
     binder
     electric-moray
@@ -72,7 +74,7 @@ for repo in "${REPOS[@]}"; do
         continue
     fi
 
-    REPO_GIT_SHA=$(git submodule $DEP_LOC | cut -c 2-41)
+    REPO_GIT_SHA=$(git submodule status $DEP_LOC | cut -c 2-41)
     if [ "$MS_GIT_SHA" == "$REPO_GIT_SHA" ]; then
         echo "$repo already has latest manta-scripts.  Not updating."
         continue
@@ -82,7 +84,7 @@ for repo in "${REPOS[@]}"; do
     git submodule init $DEP_LOC
     git submodule update $DEP_LOC
     cd $DEP_LOC
-    git pull --rebase
+    git pull --rebase origin master
     git checkout $MS_GIT_SHA
     cd -
     git add $DEP_LOC
